src/test/java/com/artipie/helm/HelmITCase.java

```diff
@@ -56,172 +34,61 @@
     private static final String CHART = "tomcat-0.4.1.tgz";
 
     /**
-     * Repo name.
+     * Test deployments.
+     * @checkstyle VisibilityModifierCheck (10 lines)
      */
-    private static final String REPO = "my-helm";
-
-    /**
-     * Temporary directory for all tests.
-     * @checkstyle VisibilityModifierCheck (3 lines)
-     */
-    @TempDir
-    Path tmp;
-
-    /**
-     * Tested Artipie server.
-     */
-    private ArtipieServer server;
-
-    /**
-     * Container.
-     */
-    private TestContainer cntn;
-
-    /**
-     * Repository url.
-     */
-    private RepositoryUrl url;
-
-    /**
-     * Artipie server port.
-     */
-    private int port;
-
-    @ParameterizedTest
-    @ValueSource(booleans = {true, false})
-    void uploadChartAndCreateIndexYaml(final boolean anonymous) throws Exception {
-        final String chartrepo = "chartrepo";
-        this.init(anonymous);
-        final HttpURLConnection con = this.putToLocalhost(anonymous);
-        MatcherAssert.assertThat(
-            "Response status is 200",
-            con.getResponseCode(),
-            new IsEqual<>(Integer.parseInt(RsStatus.OK.code()))
-        );
-        this.cntn.execStdout(
-            "helm", "init",
-            "--stable-repo-url", this.url.string(anonymous),
-            "--client-only"
-        );
-        MatcherAssert.assertThat(
-            "Chart repository was added",
-            this.helmRepoAdd(anonymous, chartrepo),
-            new StringContains(
-                String.format("\"%s\" has been added to your repositories", chartrepo)
-            )
-        );
-        con.disconnect();
-        MatcherAssert.assertThat(
-            "Repo update from chart repository",
-            this.cntn.execStdout("helm", "repo", "update"),
-            new StringContainsInOrder(
-                Arrays.asList(
-                    String.format(
-                        "Successfully got an update from the \"%s\" chart repository", chartrepo
-                    ),
-                    "Update Complete."
-                )
+    @RegisterExtension
+    final TestDeployment containers = new TestDeployment(
+        () -> TestDeployment.ArtipieContainer.defaultDefinition()
+            .withRepoConfig("helm/my-helm.yml", "my-helm"),
+        () -> new TestDeployment.ClientContainer("alpine/helm:2.16.9")
+            .withWorkingDirectory("/w")
+            .withCreateContainerCmdModifier(
+                cmd -> cmd.withEntrypoint("/bin/sh")
+                    .withCmd("-c", "while true; do sleep 300; done;")
+            ).withClasspathResourceMapping(
+                String.format("helm/%s", HelmITCase.CHART),
+                String.format("/w/%s", HelmITCase.CHART),
+                BindMode.READ_ONLY
             )
+    );
+
+    @BeforeEach
+    void setUp() throws Exception {
+        this.containers.assertExec(
```

Comment:
For setup functions you can use `container.clientExec` - it's simplified version of `assertExec` which doesn't require specifying error message and matcher.